lib/tracing-limit/src/lib.rs

```diff
@@ -381,19 +393,28 @@ impl Visit for RateLimitedSpanKeys {
 
 #[derive(Default)]
 struct LimitVisitor {
-    pub limit: Option<u64>,
+    pub limit: Option<bool>,
+    pub limit_secs: Option<u64>,
 }
 
 impl Visit for LimitVisitor {
-    fn record_u64(&mut self, field: &Field, value: u64) {
-        if self.limit.is_none() && field.name() == RATE_LIMIT_SECS_FIELD {
+    fn record_bool(&mut self, field: &Field, value: bool) {
+        if field.name() == RATE_LIMIT_FIELD {
             self.limit = Some(value);
         }
     }
 
     fn record_i64(&mut self, field: &Field, value: i64) {
-        if self.limit.is_none() && field.name() == RATE_LIMIT_SECS_FIELD {
-            self.limit = Some(u64::try_from(value).unwrap_or_default());
+        if field.name() == RATE_LIMIT_SECS_FIELD {
+            self.limit = Some(true); // limit if we have this field
+            self.limit_secs = Some(u64::try_from(value).unwrap_or_default()); // override the cli passed limit
+        }
+    }
+
+    fn record_u64(&mut self, field: &Field, value: u64) {
+        if field.name() == RATE_LIMIT_SECS_FIELD {
+            self.limit = Some(true); // limit if we have this field
+            self.limit_secs = Some(value); // override the cli passed limit
         }
     }
```

Comment:
@spencergilbert @bruceg @jszwedko let me know your thoughts about adding that new limit_secs field here.

Reply:
👍 this makes sense to me.